spark-operator/src/main/java/org/apache/spark/k8s/operator/client/RetryInterceptor.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.k8s.operator.client;
+
+import static io.fabric8.kubernetes.client.utils.Utils.closeQuietly;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import lombok.Builder;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+import org.apache.commons.lang3.StringUtils;
+
+/** Intercepts HTTP requests and add custom retry on 429 and 5xx to overcome server instability */
+@Slf4j
+@RequiredArgsConstructor
+@Builder
+public class RetryInterceptor implements Interceptor {
+  private static final String RETRY_AFTER_HEADER_NAME = "Retry-After";
+
+  private final Long maxAttemptCount;
+  private final Long maxRetryAfterInSecs;
+  private final Long defaultRetryAfterInSecs;
+
+  @Override
+  public Response intercept(Chain chain) throws IOException {
+    Request request = chain.request();
+    Response response = chain.proceed(request);
+    int tryCount = 0;
+    while (!response.isSuccessful()
+        && (response.code() == 429 || response.code() >= 500)
```

Comment:
Shall we avoid magic numbers here?

In Apache Spark repo, we used `HttpServletResponse` like the following.

https://github.com/apache/spark/blob/8ee8abaa599fd6efea85018549f1ec135af319e0/core/src/main/scala/org/apache/spark/ui/JettyUtils.scala#L82

Reply:
Response codes are hard coded as operator does not depend on servlet directly. 
After a few trading off, imo we can remove this interceptor since fabric client api is already [doing the same thing under the hood](https://github.com/fabric8io/kubernetes-client/blob/main/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/http/StandardHttpClient.java#L165-L192). This was introduced when we had to do this previously.